vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -105,12 +105,13 @@ protected void remove() {
 
     @Override
     public void selectFromClient(T item) {
-        if (isSelected(item)) {
+        Object itemId = getItemId(item);
+        if (isSelectedItemId(itemId)) {
```

Comment:
I don't see any benefit in this change, it just makes the code more convoluted, duplicates the logic from `isSelected`, and confusing because it's not clear why this doesn't simply use `isSelected`. Same with most of the other changes here that now create map instances all over the place, while sets would just do the job.

If we want to provide `getSelectedItemIds` and `isSelectedItemId` we could just add them and mention in the JavaDoc that these are exposed for sub-classes. As protected methods are considered public API, and with proper documentation, there is basically no risk of these getting removed accidentally.

Reply:
That is the intention and the reason why I have overtaken the PR from the contributor.
I will add needed notes to the documentation and clean up unnecessary changes.

Reply:
Sounds good thanks. I'll resolve this then.

Reply:
Just FYI, there is one benefit to these changes: the item ID is calculated only once. In the original code, the `isSelected(item)` will itself call `getItemId` again.

This is only a slight benefit, to be sure, but I thought to mention it.

As said, this is just FYI, I'm certainly fine keeping the original code here.